tests/cli.test.js

```diff
@@ -0,0 +1,85 @@
+// Remove double quotes around snapshots.
+// This improves snapshot readability for generated file contents.
+expect.addSnapshotSerializer({
+  test: value => typeof value === 'string',
+  print: value => value
+})
+
+jest.mock('fs')
+jest.mock('enquirer')
+
+let mockFs
+
+const runCLI = (...args) => {
+  process.exit = jest.fn()
+  console.log = jest.fn()
+  console.error = jest.fn()
+  process.cwd = jest.fn(() => '.')
+  const bin = require('../bin/actions-toolkit')
+  return bin(args)
+}
+
+beforeEach(() => {
+  jest.resetModules()
+  mockFs = require('fs')
+  mockFs.__reset()
+})
+
+test('prints help when no arguments are passed', async () => {
+  await runCLI()
+  expect(process.exit).toHaveBeenCalledWith(1)
+  expect(console.log).toHaveBeenCalledWith(
+    expect.stringMatching(/Usage: npx actions-toolkit <name>/)
+  )
+})
+
+test('prints help when --help is passed', async () => {
+  await runCLI('--help')
+  expect(process.exit).toHaveBeenCalledWith(1)
+  expect(console.log).toHaveBeenCalledWith(
+    expect.stringMatching(/Usage: npx actions-toolkit <name>/)
+  )
+})
+
+test('fails to start creating project in a directory that already exists', async () => {
+  mockFs.mkdir.mockImplementationOnce((_, cb) => {
+    const error = new Error()
+    error.code = 'EEXIST'
+    cb(error)
+  })
+
+  expect(runCLI('my-project-name')).rejects.toThrowError(
```

Comment:
I don't think I'm using the [`expect.rejects`](https://jestjs.io/docs/en/expect#rejects) API correctly here now that I take a second look. It's kind of awkward - so other ways of writing it include:

```js
return runCLI('my-project-name').catch(err => {
  expect(err).toMatch(/* ... */)
})
```

or:

```js
try {
  await runCLI('my-project-name')
} catch (err) {
  expect(err).toMatch(/* ... */)
}
```

Any preference here?

Reply:
That is actually correct - in either of the `.catch()` or `try/catch` examples, if the function _doesn't_ throw those `expect()`s will never be called and the test will still pass. The only thing is it needs to be `await`-ed:

```suggestion
  await expect(runCLI('my-project-name')).rejects.toThrowError(
```